### PR TITLE
Created AgeWarning component

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
This PR refactors the code to display an age warning, consolidating logic into one place.

Before we had both the ArticleHeadline and the MostViewed component duplicating the same html and logic, now they both import from one shared location.

```
type Props = {
    age: string;
    isScreenReader?: boolean;
    size?: 'small' | 'medium';
};
```

To prevent regression, tests have been added that show age warnings.

## Why?
For brevity

## Link to supporting Trello card
https://trello.com/c/bwYgkk3q/831-agewarning

### CODE REVIEW

File: src/web/components/ArticleHeadline.tsx [hunk 1]
```diff
@@ -1,13 +1,11 @@
 import React from 'react';
 import { css, cx } from 'emotion';
 
-import ClockIcon from '@frontend/static/icons/clock.svg';
 import { getAgeWarning } from '@root/src/lib/age-warning';
 
-import { palette } from '@guardian/src-foundations';
-import { headline, textSans } from '@guardian/src-foundations/typography';
+import { AgeWarning } from '@root/src/web/components/AgeWarning';
```
Comment (gtrufitt) [inline src/web/components/ArticleHeadline.tsx]: I think `@frontend/` rather than `@root`?

File: src/web/components/AgeWarning.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    ${isSmall ? textSans.xsmall() : textSans.medium()};
+    color: ${palette.neutral[7]};
+    background-color: ${palette.yellow.main};
+    display: inline-block;
+
+    > strong {
+        font-weight: bold;
+    }
+
+    padding: ${isSmall ? '3px 5px' : '6px 10px'};
```
Comment (gtrufitt) [inline src/web/components/AgeWarning.tsx]: Minor: These variations feel very designery-tweaky. I wonder if we could use `vw` units to make things variable but without the magic px.

File: src/web/components/ArticleHeadline.tsx (hunk 1 above)
Comment (oliverlloyd) [reply src/web/components/ArticleHeadline.tsx]: At some point Visual Code got all opinionated and we lost a lot of `@frontend` and gained a load of `@root/src`.

Shall we do a big PR to move all of these over? Arguably, we could lose `root` completely as with workspaces removed it's a bit redundant

Comment (gtrufitt) [reply src/web/components/ArticleHeadline.tsx]: Yup, makes sense to do them all in one 👍

File: src/web/components/AgeWarning.tsx (hunk 2 above)
Comment (oliverlloyd) [reply src/web/components/AgeWarning.tsx]: Yeah, I agree. At the moment I'm just copying over from frontend but this means there is a lot of magic pixel dust. I think there's also scope for another PR here to review padding and spacing in general, ideally in tandem with @SiAdcock and `src-foundation`

Comment (SiAdcock) [reply src/web/components/AgeWarning.tsx]: Our [spacing recommendations](https://zeroheight.com/2a1e5182b/p/075107) might help. Let me know if you'd like to discuss.

**Pull request merged**